### PR TITLE
Csv html exporter

### PR DESCRIPTION
**Pull request by JJJianJiao**
- implement a base Exporter class
- implement a HTMLExporter class
- Refactor CSVExporter
- Test Coverage: Yes.

### CODE REVIEW

File: Editor/Exporter.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            m_StreamWriter = new StreamWriter(path);
+        }
+
+        public void Dispose()
+        {
+            m_StreamWriter.Flush();
+            m_StreamWriter.Close();
+        }
+
+        public virtual void WriteHeader() { }
```
Comment (mtrive) [inline Editor/Exporter.cs:23]: can we make this _abstract_ since we always need an overload ?

File: Editor/Exporter.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        public void WriteIssues(ProjectIssue[] issues)
+        {
+            foreach (var issue in issues)
+                WriteIssue(issue);
+        }
+
+        protected virtual void WriteIssue(ProjectIssue issue) { }
```
Comment (mtrive) [inline Editor/Exporter.cs:31]: same here

File: Editor/HTMLExporter.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            for (var i = 0; i < m_Layout.properties.Length; i++)
+            {
+                var columnType = m_Layout.properties[i].type;
+                var prop = issue.GetProperty(columnType);
+                m_StreamWriter.WriteLine(@"<td>" + prop + @"</td>");
+            }
+            m_StreamWriter.WriteLine(@"</tr>");
+        }
+
+        public override void WriteFooter() {
```
Comment (mtrive) [inline Editor/HTMLExporter.cs:34]: curly bracket on new line

File: Tests/Editor/ProjectReportTests.cs [hunk 4]
```diff
@@ -76,19 +76,29 @@ IReadOnlyCollection<ProjectIssue> AnalyzeAndExport(IssueCategory category, strin
...
         }
 
         [Test]
-        public void ProjectReport_CodeIssues_AreExportedAndFormatted()
+        public void ProjectReport_CodeIssues_AreExportedAndFormatted_CSV()
         {
             var category = IssueCategory.Code;
             var path = string.Format("project-auditor-report-{0}.csv", category.ToString()).ToLower();
-            AnalyzeAndExport(category, path);
+            AnalyzeAndExport(category ,path, "csv");
```
Comment (mtrive) [inline Tests/Editor/ProjectReportTests.cs:101]: please remove space before comma

File: Tests/Editor/ProjectReportTests.cs [hunk 5]
```diff
@@ -164,5 +364,86 @@ public void ProjectReport_SettingsIssues_AreExportedAndFormatted()
 
             PlayerSettings.bakeCollisionMeshes = bakeCollisionMeshes;
         }
+
+        [Test]
+        public void ProjectReport_SettingsIssues_AreExportedAndFormatted_HTML() {
```
Comment (mtrive) [inline Tests/Editor/ProjectReportTests.cs:369]: curly bracket on new line

File: Tests/Editor/ProjectReportTests.cs [hunk 6]
```diff
@@ -164,5 +364,86 @@ public void ProjectReport_SettingsIssues_AreExportedAndFormatted()
...
+                        if (line.Equals("</tr>"))
+                        {
+                            index++;
+                        }
+                        if (index == 5)
+                        {
+                            issueFound = true;
+                        }
+                    }
+                    else {
```
Comment (mtrive) [inline Tests/Editor/ProjectReportTests.cs:435]: curly bracket on new line

File: Tests/Editor/ProjectReportTests.cs [hunk 7]
```diff
@@ -135,14 +244,106 @@ public void ProjectReport_CodeIssues_AreFilteredAndExported()
         }
 
         [Test]
-        public void ProjectReport_SettingsIssues_AreExportedAndFormatted()
+        public void ProjectReport_Codeissues_AreFilteredAndExported_HTML() {
```
Comment (mtrive) [inline Tests/Editor/ProjectReportTests.cs:247]: curly bracket on new line. Also _Codeissues_ should be _CodeIssues_

File: Tests/Editor/ProjectReportTests.cs [hunk 8]
```diff
@@ -135,14 +244,106 @@ public void ProjectReport_CodeIssues_AreFilteredAndExported()
...
+            var category = IssueCategory.Code;
+            var path = string.Format("project-auditor-report-{0}.html", category.ToString()).ToLower();
+            AnalyzeAndExport(category, path, "html", issue =>
+            {
+                return issue.description.StartsWith("Conversion");
```
Comment (mtrive) [inline Tests/Editor/ProjectReportTests.cs:252]: IIRC we did not want filtering in this version of the test

**Review by mtrive — CHANGES_REQUESTED** [review]
Overall looks good! Just a few minor comments.

File: Editor/Exporter.cs (hunk 1 above)
Comment (JJJianJiao) [reply Editor/Exporter.cs:23]: Good point. I did not realize it.

File: Editor/Exporter.cs (hunk 2 above)
Comment (JJJianJiao) [reply Editor/Exporter.cs:31]: Sure

File: Editor/HTMLExporter.cs (hunk 3 above)
Comment (JJJianJiao) [reply Editor/HTMLExporter.cs:34]: Sorry about my habit.

**Review by mtrive — APPROVED** [review]
Nice, thank you.

**Pull request merged**